.github/CHANGELOG.md

```diff
@@ -2,6 +2,25 @@
 
 <h3>New features since last release</h3>
 
+* Adds a device test suite, located at `pennylane/plugins/tests`, which can be used 
+  to run generic tests on core or external devices calling 
+  
+  >>> pytest pennylane/plugins/tests --device default.qubit --shots 1234 --analytic False                                                                                                                                                                                                                                                                    
```

Comment:
🎉

Reply:
I did some research on this, and it looks like this is the exact solution we eventually want: https://stackoverflow.com/questions/39091443/running-py-test-tests-from-another-pip-installed-package-against-local-code:

> I am building a package that provides (1) a fixed interface with (2) pluggable drivers. The desired driver will be selectable by the user of the package. My package will include at least one driver with it, but I would like other developers to be able to implement and validate drivers that conform to the package's interface. As such, I would like these developers to be able to run my tests against their drivers.

---

As a short term solution, we could:

* Add the function `qml.get_device_tests()` that returns the test location on the users system (similar to [`np.get_include()`](https://numpy.org/devdocs/reference/generated/numpy.get_include.html).

* Provide a convenience CLI script alongside PennyLane, that is simply a wrapper around `pytest qml.get_device_tests()`.

Reply:
I added this to the list of next milestones!